### PR TITLE
Feat/first draft

### PR DESCRIPTION
**Pull request by rodrigopavezi**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
Comment (sk91) [inline .gitignore:1]: please use full gitignore like in wallet

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      // change json to be retrieved 
+
+      let entities:Array<any> = [];  
+      entities = records.map((record:any) => {
+        return {
+          data: record.fields
+        }
+      });
+
+      res.status(200).send({entities});
```
Comment (sk91) [inline src/index.ts]: res.json is better here (that way no need to manually set json headers in responce)

also status 200 is set automatically

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  res.set("Access-Control-Allow-Credentials", "true");
+  // Send response to OPTIONS requests and terminate the function execution
+  if (req.method == 'OPTIONS') {
+    res.status(204).send('');
+  }
+
+  const tableName = req.query.tableName;
+
+  if (!tableName) {
+    res.status(400).send('Missing Table Name Query Parameter');
```
Comment (sk91) [inline src/index.ts]: should be json? reponse should be consistent

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  let records:Array<any> = [];
+  base(tableName).select().eachPage((recordsInPage: Array<any>, fetchNextPage: Function) => {
+    // merge records from all pages
+    records = [...records, ...recordsInPage];
+    fetchNextPage();
+
+  }, (err: Error) => {
+      if (err) { console.error(err); return; }
```
Comment (sk91) [inline src/index.ts]: I believe this will hang up your server, return without response

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { Request, Response } from 'express';
+import Airtable from 'airtable';
+
+const base = new Airtable({apiKey: process.env.apiKey}).base(process.env.baseId)
+
+export function airtable(req: Request, res: Response) {
+  // Set CORS headers
+  // e.g. allow GETs from https://mydomain.com with an Authorization header
+  res.set("Access-Control-Allow-Origin", "*");
```
Comment (sk91) [inline src/index.ts]: better to use cors middleware https://expressjs.com/en/resources/middleware/cors.html

here is how you can apply it:
https://codeburst.io/express-js-on-cloud-functions-for-firebase-86ed26f9144c

**Review by sk91 — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  const tableName = req.query.tableName;
+
+  if (!tableName) {
+    res.status(400).send('Missing Table Name Query Parameter');
+  }
+
+  let records:Array<any> = [];
+  base(tableName).select().eachPage((recordsInPage: Array<any>, fetchNextPage: Function) => {
```
Comment (sk91) [inline src/index.ts]: maybe extract thous callbacks to separate functions? will be cleaner that way

**Review by sk91 — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 5 above)
Comment (rodrigopavezi) [reply src/index.ts]: @sk91 I think I will stick with google docs on this https://cloud.google.com/functions/docs/writing/http. cheers

**Comment by rodrigopavezi** [comment]
@sk91 made the changes please review again. Build it failing because the credentials for google cloud does not have permissions to deploy to cloud functions.

**Review by sk91 — APPROVED** [review]

**Review by sk91 — APPROVED** [review]
Looks Good! Cheers!

**Pull request merged**